src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -179,14 +184,19 @@ public function supportsDecoding($format)
      * @param array  $result
      * @param string $keySeparator
      * @param string $parentKey
+     * @param bool   $escapeFormulas
      */
-    private function flatten(array $array, array &$result, $keySeparator, $parentKey = '')
+    private function flatten(array $array, array &$result, $keySeparator, $parentKey = '', $escapeFormulas = false)
     {
         foreach ($array as $key => $value) {
             if (is_array($value)) {
-                $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator);
+                $this->flatten($value, $result, $keySeparator, $parentKey.$key.$keySeparator, $escapeFormulas);
             } else {
-                $result[$parentKey.$key] = $value;
+                if ($escapeFormulas && \in_array(mb_substr($value, 0, 1), $this->formulasStartCharacters, true)) {
```

Comment:
What's the reason for using mb version? UTF is not used in formulasStartCharacters.